### PR TITLE
[MM-22932] - Redesign main menu categories

### PR DESCRIPTION
**Pull request by nevyangelova**
#### Summary
Redesign Main menu categories according to UX requirements

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-22932

### CODE REVIEW

File: components/widgets/menu/menu_items/menu_item.scss [hunk 1]
```diff
@@ -56,7 +56,7 @@
                 background-color: var(--center-channel-color-08) !important;
             }
         }
-
+        // TODO: add classes to avoid conflicts
```
Comment (devinbinnie) [inline components/widgets/menu/menu_items/menu_item.scss:59]: What's this TODO for? Is this a follow-up ticket or something you intend to solve now?

File: components/widgets/menu/menu_items/submenu_item.tsx [hunk 2]
```diff
@@ -41,6 +41,9 @@ type Props = {
     ariaLabel?: string;
     root?: boolean;
     show?: boolean;
+    right?: boolean;
```
Comment (devinbinnie) [inline components/widgets/menu/menu_items/submenu_item.tsx:44]: Oh no...I already wrote this code in my sidebar changes...

File: components/widgets/menu/menu_modals/submenu_modal/submenu_modal.scss [hunk 3]
```diff
@@ -2,7 +2,7 @@
 @import 'utils/variables';
 
 @media screen and (max-width: 768px) {
-
+// TODO: Let's not build CSS pyramids as it's getting hard to fight specificity. Open a Jira ticket to refactor.
```
Comment (devinbinnie) [inline components/widgets/menu/menu_modals/submenu_modal/submenu_modal.scss:5]: Same with this TODO

File: components/widgets/menu/menu_items/menu_item.scss (hunk 1 above)
Comment (nevyangelova) [reply components/widgets/menu/menu_items/menu_item.scss:59]: Ideally a follow-up ticket where the scope is known in the team as its going to be large. I would like to keep the comments and maybe add more whenever we see things that can be improved so we can refactor consistently.

File: components/widgets/menu/menu_items/submenu_item.tsx (hunk 2 above)
Comment (nevyangelova) [reply components/widgets/menu/menu_items/submenu_item.tsx:44]: 😅 I will rebase when you merge.

Comment (devinbinnie) [reply components/widgets/menu/menu_items/submenu_item.tsx:44]: Might be the other way around actually. Yours will likely get merged first, and yours also looks cleaner so I might just revert my change.

File: components/widgets/menu/menu_items/menu_item.scss (hunk 1 above)
Comment (devinbinnie) [reply components/widgets/menu/menu_items/menu_item.scss:59]: Okay, can you create the ticket and add it to this comment?

**Comment by matthewbirtch** [comment]
@nevyangelova is there a test link I can review with?

**Comment by nevyangelova** [comment]
@matthewbirtch test server created :)

**Comment by nevyangelova** [comment]
@esethna @matthewbirtch @hmhealey @devinbinnie the problem with this is that when the height is limited to 80vh (reduce screen size) we have scrolling in the menu and that limits overflow of elements, and that means that the submenu is not visible outside of the main menu (can't pop out like in the design). This is due to the structure of the menu right now, and the CSS respectively. One solution I can see is putting the submenu, next to the main menu within its container and listening for scroll, hover and click events to keep the position and visibility correct. But this will extend the Mana for this ticket significantly. Another suggestion is to somehow not scroll the menu (needs UX discussion). I am stuck on this so advice is welcome.

**Comment by jgilliam17** [comment]
Thank you @nevyangelova 
I noticed an issue with the redesigned main menu Help section on mobile web view. 
Help is missing `?` icon and opening it's sub menu off the screen (need to scroll it into view, see image)
@matthewbirtch can you please provide some guidance on what is expected in mobile web view?
I expected Help sub menu to display similarly to Account Settings

Screen Shot 2020-05-01 at 10 07 41 AM

Screen Shot 2020-05-01 at 10 57 06 AM

**Comment by matthewbirtch** [comment]
Thanks for catching this @jgilliam17. I did not account for the mobile web view in the update. That's my fault. I'm thinking something more like this:

image

image

@nevyangelova I updated the Figma file to include this. Let me know if you'd like to chat through it

**Comment by nevyangelova** [comment]
This PR is on hold because there needs to be a refactor of the component in order to accommodate this and sidebar related changes. We had a meeting with the apps team and came up with requirements and strategy and will proceed with implementation soon. Likely won't make it by the next release @matthewbirtch @esethna

**Comment by matthewbirtch** [comment]
Thanks for the update @nevyangelova. Sorry for completely missing the mobile webview side of this. It seems to be an often overlooked part of the product. We'll pick it back up when we can

**Comment by nevyangelova** [comment]
Closing this in favour of the new menu component

**Pull request closed without merging**